helix-rest/src/main/java/org/apache/helix/rest/metadatastore/ZkMetadataStoreDirectory.java

```diff
@@ -304,42 +314,35 @@ public boolean deleteShardingKey(String namespace, String realm, String sharding
    */
   @Override
   public void refreshRoutingData(String namespace) {
-    // Safe to ignore the callback if any of the maps are null.
-    // If routingDataMap is null, then it will be populated by the constructor anyway
-    // If routingDataMap is not null, then it's safe for the callback function to update it
-    if (_routingZkAddressMap == null || _realmToShardingKeysMap == null
-        || _routingDataReaderMap == null || _routingDataWriterMap == null) {
-      LOG.warn(
-          "refreshRoutingData callback called before ZKMetadataStoreDirectory was fully initialized. Skipping refresh!");
```

Comment:
This if statement shouldn't be deleted. The reasoning is clearly shown in the logging: the callback could be triggered before `ZKMetadataStoreDirectory` is fully initialized. Removing this if statement leaves the rest of the logic unguarded.

Reply:
When the `ZkMetadataStoreDirectory` is initialized, it initializes `ZkRoutingDataReader` which subscribe to ZNode changes and calls `refreshRoutingData()` when an event happens. It's possible that an event comes in before `ZkMetadataStoreDirectory` finishes initialization, which we don't want to handle. 
This is my understanding to it. This was introduced to me first by @narendly , and he was also the first to write this statement so he could provide further comments if necessary.

Reply:
@NealSun96 Thanks for the explanation.
After I check the code carefully, I don't see what you said is the case. `ZkRoutingDataReader` is initialized after the maps are created. So I think this check is safe to remove.